### PR TITLE
Blacken pulp_python

### PR DESCRIPTION
**Pull request by werwty**

### CODE REVIEW

File: pulp_python/app/serializers.py [hunk 1]
```diff
@@ -87,114 +80,136 @@ class PythonPackageContentSerializer(core_serializers.ContentSerializer):
...
-    name = serializers.CharField(
-        help_text=_('The name of the python project.')
-    )
-    version = serializers.CharField(
-        help_text=_('The packages version number.')
-    )
-    metadata_version = serializers.CharField(
-        help_text=_('Version of the file format')
+        help_text=_(
+            "The type of the distribution package " "(e.g. sdist, bdist_wheel, bdist_egg, etc)"
```
Comment (dralley) [inline pulp_python/app/serializers.py]: this is weird

File: pulp_python/app/serializers.py [hunk 2]
```diff
@@ -87,114 +80,136 @@ class PythonPackageContentSerializer(core_serializers.ContentSerializer):
...
+        required=False, allow_blank=True, help_text=_("The author's e-mail address. ")
     )
     maintainer = serializers.CharField(
-        required=False, allow_blank=True,
-        help_text=_('The maintainer\'s name at a minimum; '
-                    'additional contact information may be provided.')
+        required=False,
+        allow_blank=True,
+        help_text=_(
+            "The maintainer's name at a minimum; " "additional contact information may be provided."
```
Comment (dralley) [inline pulp_python/app/serializers.py]: here it is again

File: pulp_python/app/serializers.py [hunk 3]
```diff
@@ -87,114 +80,136 @@ class PythonPackageContentSerializer(core_serializers.ContentSerializer):
...
+        help_text=_("Text indicating the license covering the distribution"),
     )
     requires_python = serializers.CharField(
-        required=False, allow_blank=True,
-        help_text=_('The Python version(s) that the distribution is guaranteed to be '
-                    'compatible with.')
+        required=False,
+        allow_blank=True,
+        help_text=_(
+            "The Python version(s) that the distribution is guaranteed to be " "compatible with."
```
Comment (dralley) [inline pulp_python/app/serializers.py]: same here

File: pulp_python/app/serializers.py [hunk 4]
```diff
@@ -299,17 +338,19 @@ def create(self, validated_data):
...
+            specifier = python_models.ProjectSpecifier.objects.create(
+                remote=python_remote, **project
+            )
             if digests:
                 for digest in digests:
-                    python_models.DistributionDigest.objects.create(project_specifier=specifier,
-                                                                    **digest)
+                    python_models.DistributionDigest.objects.create(
+                        project_specifier=specifier, **digest
+                    )
```
Comment (dralley) [inline pulp_python/app/serializers.py]: I'm not sure I like how it does this.  It makes sense in a bunch of places but it feels wrong here.

File: pulp_python/app/serializers.py [hunk 5]
```diff
@@ -336,54 +377,57 @@ class Meta:
...
-                    detail=_('Repository has no version available to publish'))
+                    detail=_("Repository has no version available to publish")
+                )
         raise serializers.ValidationError(
-            _("Either the 'repository' or 'repository_version' need to be specified "
-              "but not both.")
+            _(
+                "Either the 'repository' or 'repository_version' need to be specified "
+                "but not both."
+            )
```
Comment (dralley) [inline pulp_python/app/serializers.py]: I don't like this

File: pulp_python/app/tasks/publish.py [hunk 6]
```diff
@@ -117,27 +120,25 @@ def write_simple_api(publication):
...
-        metadata_relative_path = '{project_dir}/index.html'.format(project_dir=project_dir)
+        metadata_relative_path = "{project_dir}/index.html".format(project_dir=project_dir)
 
-        with open(metadata_relative_path, 'w') as simple_metadata:
-            context = Context({
-                'project_name': name,
-                'project_packages': package_detail_data
-            })
+        with open(metadata_relative_path, "w") as simple_metadata:
+            context = Context({"project_name": name, "project_packages": package_detail_data})
```
Comment (dralley) [inline pulp_python/app/tasks/publish.py]: This looks nicer on two lines

File: pulp_python/app/tasks/sync.py [hunk 7]
```diff
@@ -73,7 +73,8 @@ def _fetch_inventory(version):
     inventory = set()
     if version:
         for content in python_models.PythonPackageContent.objects.filter(
-                pk__in=version.content).only("filename"):
+            pk__in=version.content
+        ).only("filename"):
```
Comment (dralley) [inline pulp_python/app/tasks/sync.py]: meh

File: pulp_python/tests/functional/api/test_crud_publishers.py [hunk 8]
```diff
@@ -39,48 +39,46 @@ def test_01_create_publisher(self):
...
-    @selectors.skip_if(bool, 'publisher', False)
+    @selectors.skip_if(bool, "publisher", False)
     def test_02_read_publishers(self):
         """Read a publisher by its name."""
-        page = self.client.get(PYTHON_PUBLISHER_PATH, params={
-            'name': self.publisher['name']
-        })
-        self.assertEqual(len(page['results']), 1)
+        page = self.client.get(PYTHON_PUBLISHER_PATH, params={"name": self.publisher["name"]})
+        self.assertEqual(len(page["results"]), 1)
```
Comment (dralley) [inline pulp_python/tests/functional/api/test_crud_publishers.py]: A lot of these tests look better on two lines

**Comment by werwty** [comment]
XD we're too opinionated for black

And it does try to format this line https://github.com/pulp/pulp_python/pull/177/files#diff-27efe7f2ffa6f1561afb34c9ffa703faR257
to be `for repo_version in self.repo_versions[index + 1 :]:` which is not pep8 compliant 

It was worth a try, but black's probably not quite right for us right now

**Comment by dralley** [comment]
I'm not sure how I feel about it.  I agree with some of the things it did and disagree with a lot of others.  I don't really like the idea of formatting the smash tests because I want to be able to easily compare them against the file smash tests.

**Comment by werwty** [comment]
TBF we can point it only to pulp_python/app and not have it touch the smash tests. 
I do like a lot of the changes it made, and I like the idea of auto code formatting. But I'm not 100% loving black's style

**Comment by dralley** [comment]
I wouldn't mind taking "the good suggestions" and making a formatting PR for them

**Comment by werwty** [comment]
That's not a bad idea, I can modify this PR to be "good" formatting only.
We do lose a lot of the power black brings to the table, but I can keep looking for good auto formatting tools.

**Comment by dralley** [comment]
I think at a minimum we can autoformat outright PEP8 errors, and I know tools for that exist.

**Comment by dralley** [comment]
Apparently ```@pep8speaks pep8ify``` will make a PR against your PR branch which

**Comment by dralley** [comment]
Wrong button sorry

**Pull request closed without merging**